### ISSUE TITLE
Division by zero in log message

### ISSUE THREAD

**Issue by koelling** [CGATOxford/UMI-tools#190]
This is a bit of a weird edge case, but I was trying to run umi_tools dedup on a BAM file that only contained a single read pair, of which the second read is unmapped, and got this error:

```
Traceback (most recent call last):
  File "bin/umi_tools", line 11, in <module>
    load_entry_point('umi-tools==0.5.0', 'console_scripts', 'umi_tools')()
  File "lib/python3.5/site-packages/umi_tools/umi_tools.py", line 50, in main
    module.main(sys.argv)
  File "lib/python3.5/site-packages/umi_tools/dedup.py", line 391, in main
    processor.UMIClusterer.positions))
ZeroDivisionError: float division by zero
```

I think this is caused by this line, where you divide by processor.UMIClusterer.positions, which is zero in this case: https://github.com/CGATOxford/UMI-tools/blob/master/umi_tools/dedup.py#L390

**Comment by IanSudbery** [comment]
Good catch. Should be an easy fix. I guess.

**Comment by slebedeva** [comment]
Got the same error when working with test data (just a small part of the original fastq).

**Comment by TomSmithCGAT** [comment]
Hi @koelling & @slebedeva - Would you mind installing from the {TS}-Debug_no_positions branch to confirm this bug is resolved on the branch?

**Comment by slebedeva** [comment]
Confirmed - the bug is gone for me! Thanks a lot!!!

**Comment by TomSmithCGAT** [comment]
Brilliant. I'll merge this branch with master and release 0.5.1 shortly

**Comment by TomSmithCGAT** [comment]
Resolved in version [0.5.1](https://github.com/CGATOxford/UMI-tools/releases/tag/0.5.1)
